File: clients/FSharp/Model/BuildAction.fs
#nowarn "0058"
namespace Aicup2020.Model
type BuildAction = {
    EntityType: EntityType;
    Position: Vec2Int;
    } with
    member this.writeTo(writer: System.IO.BinaryWriter) =
        writer.Write (int this.EntityType)
        this.Position.writeTo writer
    static member readFrom(reader: System.IO.BinaryReader) = {
        EntityType = reader.ReadInt32() |> enum
        Position = Vec2Int.readFrom reader
    }

File: clients/Scala/src/main/scala/model/BuildAction.scala
package model

import util.StreamUtil

case class BuildAction(entityType: model.EntityType, position: model.Vec2Int) {
    def writeTo(stream: java.io.OutputStream) {
        entityType.writeTo(stream)
        position.writeTo(stream)
    }
}
object BuildAction {
    def readFrom(stream: java.io.InputStream): BuildAction = BuildAction(
        model.EntityType.readFrom(stream)
        ,
        model.Vec2Int.readFrom(stream)
        )
}
